ui/src/variables/reducers/index.ts

```diff
@@ -81,13 +82,29 @@ export const variablesReducer = (
 
       case 'SET_VARIABLE_VALUES': {
         const {contextID, status, values} = action.payload
+        const originalOrder = draftState.values[contextID].order
```

Comment:
This line was preventing me from loading a dashboard that had never been loaded before. In this case there are no values in localStorage, so `draftState.values[contextID]` doesn't exist, so accessing its `order` property is a run time error.

I think we can also simplify the logic that maintains the existing order:

```
        const prevOrder = get(draftState, `values.${contextID}.order`) || []

        if (values) {
          // Maintain the existing sort order, placing new variables at the
          // beginning
          const order = Object.keys(values).sort(
            (a, b) => prevOrder.indexOf(a) - prevOrder.indexOf(b)
          )

          draftState.values[contextID] = {
            status,
            values,
            order,
          }
        } else if (draftState.values[contextID]) {
```

Reply:
clever! I like it